### PR TITLE
fix: some apis bypassed authentication

### LINKED ISSUE #1351: bug: some apis bypassed authentication
# Bug report

## Describe the bug

I found that some apis that need to get the original data of the request without using the droplet bypass the authentication, I think it's due to #1296

we need to fix it before release.

## System information

- OS: [e.g. macOS, Windows] macOS
- Browser (if applies) [e.g. Chrome, Safari, Edge]
- Version: [e.g. 2.0-rc3] master

### PR DESCRIPTION
**Pull request by nic-chen**
Please answer these questions before submitting a pull request

- Why submit this pull request?
- [x] Bugfix
- [ ] New feature provided
- [ ] Improve performance
- [ ] Backport patches

- Related issues
close #1351

___
### Bugfix
- Description

I found that some apis that need to get the original data of the request without using the droplet bypass the authentication, I think it's due to #1296

- How to fix?

upgrade `droplet` framework, and update the apis that not using `droplet` to use `droplet`

### CODE REVIEW

**Review by tokers — APPROVED** [review]

File: api/test/e2e/base.go [hunk 1]
```diff
@@ -54,6 +54,7 @@ func init() {
 
 	url := ManagerAPIHost + "/apisix/admin/user/login"
 	req, err := http.NewRequest(http.MethodPost, url, bytes.NewBuffer(requestBody))
+	req.Header.Add("Content-Type", "application/json")
```
Comment (starsz) [inline api/test/e2e/base.go:57]: Just to make sure that FE will add this header when query the BE.

@nic-chen @juzhiyuan

Comment (juzhiyuan) [reply api/test/e2e/base.go:57]: ya

File: api/test/docker/Dockerfile [hunk 2]
```diff
@@ -23,8 +23,6 @@ COPY ./ ./
 
 RUN mkdir -p /go/manager-api/conf \
     && mkdir -p /go/manager-api/build-tools \
-    && go env -w GOPROXY=https://goproxy.io,direct \
```
Comment (imjoey) [inline api/test/docker/Dockerfile:26]: @nic-chen I was wondering that is removing Go proxy related to this PR ? Thanks.

**Review by imjoey — CHANGES_REQUESTED** [review]
Everything looks good to me, except some tiny modifications. Thank you.

File: api/go.sum [hunk 3]
```diff
@@ -339,6 +339,8 @@ github.com/satori/go.uuid v1.2.0/go.mod h1:dA0hQrYB0VpLJoorglMZABFdXlWrHn1NEOzdh
 github.com/sean-/seed v0.0.0-20170313163322-e2103e2c3529/go.mod h1:DxrIzT+xaE7yg65j358z/aeFdxmN0P9QXhEzd20vsDc=
 github.com/shiningrush/droplet v0.2.4 h1:OW4Pp+dXs9O61QKTiYSRWCdQeOyzO1n9h+i2PDJ5DK0=
 github.com/shiningrush/droplet v0.2.4/go.mod h1:akW2vIeamvMD6zj6wIBfzYn6StGXBxwlW3gA+hcHu5M=
```
[the remark this answers is not in the document]
Comment (nic-chen) [reply api/go.sum:341]: @imjoey   lib `github.com/shiningrush/droplet/wrapper/gin` using `github.com/shiningrush/droplet v0.2.4`

[the remark this answers is not in the document]
Comment (imjoey) [reply api/go.sum:341]: > @imjoey lib `github.com/shiningrush/droplet/wrapper/gin` using `github.com/shiningrush/droplet v0.2.4`

@nic-chen okey, I see. Will mark this resolved. Thanks.

File: api/test/docker/Dockerfile (hunk 2 above)
Comment (nic-chen) [reply api/test/docker/Dockerfile:26]: we don't need it in CI. @imjoey

Comment (membphis) [reply api/test/docker/Dockerfile:26]: ping @imjoey

Comment (imjoey) [reply api/test/docker/Dockerfile:26]: > we don't need it in CI. @imjoey

@nic-chen thanks. It seems that we could also remove it in another PR. We'd better keep one PR only do one thing then. 😃 Thanks.

**Review by imjoey — APPROVED** [review]
/lgtm

Will merge this once CI passes. Thanks @nic-chen .

**Review by starsz — APPROVED** [review]

**Pull request merged**